src/main/fc/rc.c

```diff
@@ -572,55 +567,66 @@ FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, int axis)
     // smooth the setpointSpeed value
     setpointSpeed = prevSetpointSpeed[axis] + pid->feedforwardSmoothFactor * (setpointSpeed - prevSetpointSpeed[axis]);
 
-    // calculate acceleration and attenuate
+    // calculate acceleration from smoothed setpointSpeed and attenuate
     setpointAcceleration = (setpointSpeed - prevSetpointSpeed[axis]) * rxRate * 0.01f;
     prevSetpointSpeed[axis] = setpointSpeed;
 
-    // smooth the acceleration element (effectively a second order filter) and apply jitter reduction
+    // smooth the acceleration element (effectively a second order filter since incoming setpoint was smoothed)
+    // apply jitter reduction to acceleration here, effectively twice on acceleration
     setpointAcceleration = prevAcceleration[axis] + pid->feedforwardSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
     prevAcceleration[axis] = setpointAcceleration * zeroTheAcceleration;
     setpointAcceleration = setpointAcceleration * pid->feedforwardBoostFactor * jitterAttenuator * zeroTheAcceleration;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 1, lrintf(setpointSpeed * 0.1f)); // base feedforward without acceleration
-    }
+    feedforward = setpointSpeed + setpointAcceleration;
 
-    float feedforward = setpointSpeed + setpointAcceleration;
+    // apply jitter attenuation to classic feedforward elements only (twice on acceleaertion)
+    feedforward *= jitterAttenuator;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 2, lrintf(feedforward * 0.1f));
-        // un-smoothed feedforward including acceleration but before limiting, transition, averaging, and jitter reduction
-    }
-
-    // apply feedforward transition
+    // apply feedforward transition to classic feedforward elements only
     const bool useTransition = (pid->feedforwardTransition != 0.0f) && (rcDeflectionAbs[axis] < pid->feedforwardTransition);
     if (useTransition) {
         feedforward *= rcDeflectionAbs[axis] * pid->feedforwardTransitionInv;
     }
 
-    // apply averaging
-    if (feedforwardAveraging) {
-        feedforward = laggedMovingAverageUpdate(&feedforwardDeltaAvg[axis].filter, feedforward);
+    // apply max rate limiting, forcing to zero when sticks approach max, but not on yaw
+    if (axis < FD_YAW) {
+        if (pid->feedforwardMaxRateLimit) {
+            if (feedforward * setpoint > 0.0f) { // in same direction
+                const float limit = (maxRcRate[axis] - fabsf(setpoint)) * pid->feedforwardMaxRateLimit;
+                feedforward = (limit > 0.0f) ? constrainf(feedforward, -limit, limit) : 0.0f;
+            }
+        }
+    } else {
+        // add high-pass filtered setpoint to feedforward for yaw
+        // this is not a derivative, being proportional to setpoint, so does not require jitter reduction
```

Comment:
(derivative is a kind of high-pass filter)

Can you try to reword this paragraph?

Reply:
Thanks for this comment.
You are indeed right, a high-pass is similar to a derivative, especially a 'fast' high-pass, where the high-pass is very similar to simple derivative.
However, this is such a 'slow' high-pass, that it still climbs, especially after a fast stick movement with a quick new setpoint, after the setpoint is constant.  And it falls slowly while the sticks stay constant, so it's not really a classic derivative, nor is it even a low-passed classic derivative.
I'm not sure really what it is :-)
I'll modify the comments which, I agree, are unhelpful.